devito/data/decomposition.py

```diff
@@ -239,30 +256,32 @@ def index_glb_to_loc(self, *args, rel=True):
                 elif isinstance(glb_idx, slice):
                     if self.loc_empty:
                         return slice(-1, -3)
-                    if glb_idx.step >= 0 and glb_idx.stop == self.glb_min:
-                        glb_idx_min = self.glb_min if glb_idx.start is None \
+                    if glb_idx.step >= 0 and glb_idx.stop == glb_min:
+                        glb_idx_min = glb_min if glb_idx.start is None \
                             else glb_idx.start
-                        glb_idx_max = self.glb_min
+                        glb_idx_max = glb_min
                         retfunc = lambda a, b: slice(a, b, glb_idx.step)
                     elif glb_idx.step >= 0:
```

Comment:
Can this be done in a better way? i.e. this is the same condition w line 259?

Reply:
@JDBetteridge was taking a look at it last week. Should definitely be done, but in a separate PR